stranding/stranding.py

```diff
@@ -0,0 +1,190 @@
+            return 0
+        return alignment
+
+    def align_and_log(self, ref, query):
+        alignments = self.align(ref, query, False)
+        for a in alignments:
+            if self.is_high_scoring(a, query):
+                LOGGER.error(format_alignment(*a))
+
+    def strand_flanks(self, _5p, _3p, build, chr_name, pos, window=DEFAULT_WINDOW_EXTENSION):
+        """
+        This is a flank stranding algorithm for sequences mapped to a human genome
+        reference assembly. Mapping coordinates are required! This is not BLAT or BLAST.
+
+        Given one or both flanks and genome mapping coordinates it determines
+        if the flanking sequence(s) coorespond to the forward or reverse strand of the
+        specified reference assembly.
+
+        It can optionally look beyond exact mapping coordinates to search nearby regions
+        (up to the `window` size specified) but takes longer as local alignments are
+        expensive on long sequences.
+
+        The `tolerance` setting defines the minimum alignment score relative to the
+        query sequence length. This is also impacted by changes to the alignment
+        scoring parameters
+
+        When `tolerance` is 1.0 and `window` is 0.0 the algorithm will only check for
+        exact sequence matches at the specified coordinates. This is the most performant
+        use case as no alignments are performed.
+
+        Otherwise, the algorithm will load the reference sequences for the 5' and 3'
+        flanks at the specified coordinates extending in each direction extended by
+        `window`. These sequences and their reverse complements are aligned and
+        scored against the query flanks. Alignments scoring above
+        `len(query flank) * match_score * tolerance` are accepted.
+        (a perfect alignment has a score of `len(query flank) * match_score`)
+
+        A return value of 1 indicates that alignments were accepted against the forward
+        reference sequence and the flanks are on the forward strand of the specified
+        reference assembly.
+
+        A return value of -1 indicates that alignments were accepted against the
+        reverse complement of the forward reference sequence and the flanks correspond
+        to the "reverse" or "minus" strand of the specified reference assembly.
+
+        An InconsistentAlignment exception is raised if alignments are accepted on
+        both strands. An Unstrandable exception is raised if no alignments are accepted.
+        """
+        # sanity checks
+        if pos == 0:
+            raise Unstrandable('Position 0 is unmapped')
+        elif chr_name in ('0', 0):
+            raise Unstrandable('Chromosome 0 is unmapped')
+        elif max(len(_5p), len(_3p)) < self.min_flank_length:
+            raise FlanksTooShort('At least one flank must be longer than the specified'
+                                 'minimum flank length of s %d' % self.min_flank_length)
```

Comment:
* missing space between 'specified' and 'minimum'
* extraneous 's' between 'of' and '%d'